client/src/components/generic/Table.js

```diff
@@ -47,15 +47,29 @@ const StyledTableRow = withStyles((theme) => ({
   },
 }))(TableRow);
 
-export const Table = ({ order, orderBy, onRequestSort, columns, rows, isLoading, rowsPerPage = 10 }) => {
+export const Table = ({ order, orderBy, renderObjectCell, onRequestSort, columns, rows, isLoading, rowsPerPage = 10 }) => {
 
   const [pageRows, setPageRows] = useState([]);
   const [page, setPage] = useState(0);
 
   useEffect(() => {
     const offset = page * rowsPerPage;
-    setPageRows(rows.slice(offset, offset + rowsPerPage))
-  }, [rows, page, rowsPerPage]);
+
+    const paginatedRows = rows.slice(offset, offset + rowsPerPage);
+
+    if (renderObjectCell) {
```

Comment:
This feels pretty weird. But we've got a deadline so if it works, so be it.

Reply:
Indeed, I found a better and more readable way to do that.